src/parsers/power_system_table_data.jl

```diff
@@ -807,99 +807,160 @@ function make_generator(data::PowerSystemTableData, gen, cost_colnames, bus, gen
     return generator
 end
 
-function calculate_variable_cost(
-    data::PowerSystemTableData,
+function make_cost(
+    ::Type{T},
+    data,
     gen,
     cost_colnames::_HeatRateColumns,
-    base_power,
-)
-    fuel_cost = gen.fuel_price / 1000.0
+) where {T <: ThermalGen}
+    fuel_price = gen.fuel_price / 1000.0
 
-    vom = isnothing(gen.variable_cost) ? 0.0 : gen.variable_cost
+    cost_pairs = get_cost_pairs(gen, cost_colnames)
+    var_cost, fixed = create_pwinc_cost(cost_pairs)
 
-    if fuel_cost > 0.0
-        var_cost =
-            [(getfield(gen, hr), getfield(gen, mw)) for (hr, mw) in cost_colnames.columns]
-        var_cost = unique([
-            (tryparse(Float64, string(c[1])), tryparse(Float64, string(c[2]))) for
-            c in var_cost if !in(nothing, c)
-        ])
-        if isempty(var_cost)
-            @warn "Unable to calculate variable cost for $(gen.name)" var_cost maxlog = 5
-        end
-    else
-        var_cost = [(0.0, 0.0)]
-    end
+    startup_cost, shutdown_cost = calculate_uc_cost(data, gen, fuel_price)
 
-    if length(var_cost) > 1
-        var_cost[2:end] = [
-            (
-                (
-                    var_cost[i][1] * fuel_cost * (var_cost[i][2] - var_cost[i - 1][2]) +
-                    var_cost[i][2] * vom
-                ),
-                var_cost[i][2],
-            ) .* gen.active_power_limits_max .* base_power for i in 2:length(var_cost)
-        ]
-        var_cost[1] =
-            ((var_cost[1][1] * fuel_cost + vom) * var_cost[1][2], var_cost[1][2]) .*
-            gen.active_power_limits_max .* base_power
-
-        fixed = max(
-            0.0,
-            var_cost[1][1] -
-            (var_cost[2][1] / (var_cost[2][2] - var_cost[1][2]) * var_cost[1][2]),
-        )
-        var_cost[1] = (var_cost[1][1] - fixed, var_cost[1][2])
+    op_cost = ThermalGenerationCost(
+        FuelCurve(var_cost, UnitSystem.NATURAL_UNITS, fuel_price),
+        fixed * fuel_price,
+        startup_cost,
+        shutdown_cost,
+    )
+    return op_cost
+end
+
+function make_cost(
+    ::Type{T},
+    data,
+    gen,
+    cost_colnames::_CostPointColumns,
+) where {T <: ThermalGen}
+    cost_pairs = get_cost_pairs(gen, cost_colnames)
+    var_cost = create_pwl_cost(cost_pairs)
```

Comment:
Is it the case that when the input data is a fuel curve, it's given as incremental/marginal, whereas when the input data is a cost curve, it's given as input-output? That seems plausible, just making sure. If so, maybe put in a comment to that effect.

Reply:
yes. the cost-point data option is less defined and less exercised.

Reply:
(see above my question regarding why we decided to make cost curves input-output when fuel curves are incremental)